### ISSUE TITLE
eBMC GUI: Network Settings

### ISSUE THREAD

**Issue by gtmills** [ibm-openbmc/dev#407]
## Stakeholders
**SME**: @ratagupt 
**PHYP SME:** Dave Stanton, Tom Burks
**DR**: @nicoleconser 
**UX**: @jandraa 
**FED**: @dixsie 

## InVision Prototype
https://ibm.invisionapp.com/share/8ENYRVXAPFD#/319115961_Physical

## Design Issue
* IPV6: openbmc/phosphor-webui/issues/62
* openbmc/phosphor-webui#7

## Development Issue
* **Error Instances**: https://github.com/openbmc/phosphor-webui/issues/14
* **Feature Updates**: https://github.com/openbmc/phosphor-webui/issues/15

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by jandraa** [comment]
Pending Questions  -- I have reached out to Ratan, Prashanth, Raviteja, and Sunitha for help answering these questions. 

**MAC Address**
- [ ] Should MAC Addresses be configurable / reset to factory default? 
    - We've heard: Usually users don't need to change it, but there are cases when they have to. I'm not sure that the original MAC or even the state of it being overridden will be available via RedFish, but it would really be good to have it. Some "Reset MAC to factory default" function needs to be there. At least as a checkbox that would be sent to backend that would then reset the MAC address.
---
**Hostname**
- [ ] Verify that a hostname can be variable from interface to interface.
     - Ratan believes the hostname is for the system.
     - A couple users have mentioned the server name is separate and they have a need for changing the hostname on an interface basis. 
---
**IP Addresses**
- [ ] Is there a way to list both subnet mask (IPV4) and prefix length (IPV6) with a single term? 
---
**VLAN**
- [ ] How should VLANs and VLAN IDs be represented on a physical interface settings page? on a VLAN interface settings page? Does the VLAN section need to look different for the settings page of the two types of interfaces?
- [ ] When creating a new VLAN, should users be able to assign multiple physical 'master' interfaces/ports?
- [ ] When/how does a VLAN interface get a MAC address (is it inherited)?
---
**DHCP configurations**
- [ ] How does a change in the DHCP config popup relate to enabling/disabling DHCP for IPV4 or IPV6 of a particular interface?
- [ ] Does there need to be two DHCP config popup modules, one for DHCPv4 and one for DHCPv6? 
- [ ] Is this where the DUID should live?

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by jandraa** [comment]
@ratagupt please advice on the questions above, particularly the bolded questions about MAC addresses (local or not) being editable per interface through the GUI, hostnames being editable per interface through the GUI, and the different needs for the VLAN section on the settings page of physical interfaces and VLAN interfaces.

**Comment by jandraa** [comment]
@nicoleconser Need user research on: 

- The degree of flexibility users want with IP addresses
-- how many
-- both static and dynamic at the same time (use case) 

- How/when do they interact with VMI? (VMI and BMC on the same page or different)

**Comment by rfrandse** [comment]
refresh

**Comment by derick-montague** [comment]
This issue will also include IPV6 now that that support need has been pushed out of milestone 9.4.1.

**Comment by derick-montague** [comment]
@jandraa are you still waiting on @ratagupt to follow up from the questions you asked above?

**Comment by jandraa** [comment]
Yes, for the most part.

**Should MAC Addresses be configurable / reset to factory default? Is this the local MAC address?**
This one I don't have much clarity one yet. It seems clear that we would want changing the MAC address to be for super admin privileges only, but I am unsure if that means it shouldn't be the the GUI or it can be because this panel would only be accessible to those with super admin privileges. @ratagupt @geissonator what are your thoughts? 

>Is there a local hostname for each interface or an interface name other than the 'eth01' names?

Yes, each interface has a Fully Qualified Domain Name (FQDN). Aside from that, the system has one single host name. 

>Are there different needs for the VLAN section on the settings page of the two types of interfaces (physical interfaces and VLAN interfaces)? If so, how and what are they?

Yes, the virtual interface pages shouldn't have a VLAN section.

>Does there need to be two DHCP config popup modules, one for DHCPv4 and one for DHCPv6?

One for both.

>Is this where the DUID should live?

Yes, DUID would live in the DHCP config popup.

**Comment by jandraa** [comment]
The questions now are dealing more with Network Settings in relation to the VMI or at least have come up in conversations with them. The questions for that are: 
- [ ] Should the VMI and BMC Network settings be in one single panel or should they each have their own?
- [ ] Are we supporting dynamic and static IP addresses coexisting and how many do we want to support? 
-- Technically we can do it but we don't know why users would use it. 
- [ ] Zero-config, what is the use case for VMI and how does that impact the panel, if at all? 
- [ ] What is the use case for more than one VLAN ID?
-- Technically we can do it but we don't know why users would use it.

**Comment by rfrandse** [comment]
refresh

**Comment by derick-montague** [comment]
## Multiple VLAN IDs
- OpenPower systems: VLAN section will be included in the GUI. Users will have the flexibility to configure multiple VLANs; if they don't have a use case for it, they can choose not to.
- Enterprise systems: There is no requirement for VLANs on enterprise systems so the VLAN section will not be shown in this version of the GUI.
 
## Multiple IP addresses 
- OpenPower systems: There will be no limit on the number of IP addresses on a single interface 
- Enterprise systems: We will limit to 1 IP address for BMC and 1 for VMI (on a single interface) because there is no use case for multiple and it would drive unnecessary complexity.
 
## DHCP and static addresses coexisting 
- OpenPower systems: Will be supported, although users would likely only configure with one or the other.
- Enterprise systems: Both IP addresses (BMC and VMI) will need to come from the same source, either DHCP or static. Will not support both coexisting within a single interface at the same time. 
 
## Zero-config
- OpenPower systems: Will be supported. 
- Enterprise systems: Not supported; no use case or need.

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by mzipse** [comment]
@derick-montague  I removed the "Not Applicable" Milestone as adding IPV6 to the GUI is real work for eBMC.  If this is already covered elsewhere should we close this.

**Comment by derick-montague** [comment]
We closed the IPv6 story quite a while ago. Technically, this epic is complete since all stories are closed. The network settings page that @miramurali23 created was never merged due to outstanding backend reviews, so we were planning to abandon that review. 

The IPv6 story was closed due to lack of user support and we discussed pushing it out. If that's, changed, I will need to find the backend stories supporting the change, however, we had a design supporting IPv6. I briefly remember changing the milestone due to 9.5.2 not being relevant.

**Comment by derick-montague** [comment]
@mzipse - GUI: Add IPV6 to network page #408 linked above had the `invalid` label added to it two days ago. We are working on the Network page in our current sprint, milestone `Sprint 2020-05`. Do we know if the backend is complete? I believe we can close this epic.

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh
